lib/Bindings/Python/dialects/om.py

```diff
@@ -0,0 +1,89 @@
+#  Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+#  See https://llvm.org/LICENSE.txt for license information.
+#  SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+from ._om_ops_gen import *
+from .._mlir_libs._circt._om import Evaluator as BaseEvaluator, Object
+
+from circt.ir import Attribute, Diagnostic, DiagnosticSeverity, Module, StringAttr
+from circt.support import attribute_to_var, var_to_attribute
+
+import sys
+import logging
+from dataclasses import fields
+from typing import TYPE_CHECKING, Any, TypeVar
+
+if TYPE_CHECKING:
+  from _typeshed.stdlib.dataclass import DataclassInstance
+
+
+# Define the Evaluator class by inheriting from the base implementation in C++.
+class Evaluator(BaseEvaluator):
+
+  def __init__(self, mod: Module) -> None:
+    """Instantiate an Evaluator with a Module."""
+
+    # Call the base constructor.
+    super().__init__(mod)
+
+    # Set up logging for diagnostics.
+    logging.basicConfig(
+        format="[%(asctime)s] %(name)s (%(levelname)s) %(message)s",
+        datefmt="%Y-%m-%d %H:%M:%S",
+        level=logging.INFO,
+        stream=sys.stdout,
+    )
+    self._logger = logging.getLogger("Evaluator")
+
+    # Attach our Diagnostic handler.
+    mod.context.attach_diagnostic_handler(self._handle_diagnostic)
+
+  def instantiate(self, cls: type["DataclassInstance"],
+                  *args: Any) -> "DataclassInstance":
```

Comment:
What is the indented interface of `instantiate`? 
Currently we can pass variadic arguments to `instantiate` like `evaluator.instantiate(Test, 42, 43, 45)` but it fails while evaluation (specifically at `var_to_attribute(*args)` as `*args` is unpacked into arguments). I think we have to change the signature into
`def instantiate(self, cls: type["DataclassInstance"], List[Any]` or to fix the implementation to accept variadic arguments.

Reply:
Oops, I will add a test, my intent was for the variadic API, and to pass the list of values to `var_to_attribute`, which will make Attributes out of each element and return an ArrayAttr containing them.

Reply:
That makes sense!